modules/sdk-core/src/bitgo/lightning/lightning.ts

```diff
@@ -114,4 +118,32 @@ export class Lightning implements ILightning {
       throw new Error(`error(s) parsing response body: ${errors}`);
     });
   }
+
+  public async decodeLnurlPay(lnurl: string): Promise<LnurlPayResponse & { domain: string }> {
+    const url = decodeLnurl(lnurl);
+    const body = await this.bitgo.get(url).result();
+    const decodedRes = decodeOrElse(LnurlPayResponse.name, LnurlPayResponse, body, (errors) => {
+      throw new Error(`error(s) parsing lnurl response: ${errors}`);
+    });
+    const { availableBalance } = await this.getBalance();
```

Comment:
I don't think we want to look up our available balance here. That's hitting the bitgo API which again isn't ideal for internal use. And regardless of that, our balance is subject to change and doesn't need to be a limiting factor here. Let's keep this client side and just use the `maxSpendable`.

Reply:
so regardless of user' balance, we should let the user be able to make lnurl pay request, and later in wp, if the user has no sufficient balance, it then throws error?